### PR TITLE
Update documentation with new knowledge

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Update README with info on how to write sanitary backend tests (we should build upon this) and how to trigger a DB backup from the CircleCI UI.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-0

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by AdamAdHocTeam — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -108,6 +108,16 @@ On the frontend, the lcov and HTML files are generated as normal, however on the
...
+It's important that our tests fully clean up after themselves if they interact with the database. This way, tests do not conflict when run on the CI and remain as deterministic as possible.The best way to do this is to run them locally in an isolated environment and confirm that they are sanitary. 
+
+With that in mind, there a few "gotchas" to remember to help write sanitary tests.
+- ```Grant.destroy``` needs to run with ```individualHooks: true``` or the related GrantNumberLink model prevents delete.
+- ```Model.destroy``` should run  ```individualHooks: true``` by default in your tests. There may be times when this is undesirable; this should be indicated with a comment
+- Be aware of paranoid models.  For those models: force: true gets around the soft delete. If they are already soft-deleted though, you need to remove the default scopes paranoid: true does it, as well as Model.unscoped() 
+- There are excellent helpers for creating and destroying common Model mocks in ```testUtils.js```. Be aware that they take a scorched earth approach to cleanup. For example, when debugging a flaky test, it was discovered that ```destroyReport``` was removing a commonly used region.
+
```
Comment (GarrettEHill) [inline README.md:119]: programmatic transaction is already and could be referenced here

File: README.md [hunk 2]
```diff
@@ -417,6 +427,9 @@ Our project includes four deployed Postgres databases, one to interact with each
     cf disallow-space-ssh ttahub-prod
     ```
 
+### Taking a production backup via CircleCI
+We can quickly take a production backup via the CircleCI web interface. To do so, go to the ```production``` branch there and trigger a pipeline with the variable ```manual-trigger``` set to true. You can then retrieve this backup with the script ```bin/latest_backup.sh```.
+
```
Comment (GarrettEHill) [inline README.md:432]: thanks for adding this

**Review by GarrettEHill — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -108,6 +108,16 @@ On the frontend, the lcov and HTML files are generated as normal, however on the
...
 
+### Helpful notes on writing (backend) tests
+It's important that our tests fully clean up after themselves if they interact with the database. This way, tests do not conflict when run on the CI and remain as deterministic as possible.The best way to do this is to run them locally in an isolated environment and confirm that they are sanitary. 
+
+With that in mind, there a few "gotchas" to remember to help write sanitary tests.
+- ```Grant.destroy``` needs to run with ```individualHooks: true``` or the related GrantNumberLink model prevents delete.
+- ```Model.destroy``` should run  ```individualHooks: true``` by default in your tests. There may be times when this is undesirable; this should be indicated with a comment
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply README.md:116]: Right, that's what I was trying to say. Will revise to make it clearer.

File: README.md [hunk 4]
```diff
@@ -108,6 +108,16 @@ On the frontend, the lcov and HTML files are generated as normal, however on the
...
+- Be aware of paranoid models.  For those models: force: true gets around the soft delete. If they are already soft-deleted though, you need to remove the default scopes paranoid: true does it, as well as Model.unscoped()
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply README.md:117]: Unless I'm mistaken, ```force: true``` does a true delete but it does not ignore scopes (if a goal was already soft-deleted, ```force: true``` doesn't do anything)

**Pull request merged**